packages/openactive-broker-microservice/test/util/item-transforms-test.js

```diff
@@ -0,0 +1,102 @@
+const { expect } = require('chai');
+const {invertFacilityUseItems, createItemFromSubEvent} = require('../../src/util/item-transforms');
+
+describe('test/utils/item-transforms-test', () => {
+  describe('invertFacilityUseItems', () => {
+    it('should return the input items when no invertibleFacilityUseItems are present', () => {
+      const items = [
+        { data: { individualFacilityUse: null } },
+        { data: { individualFacilityUse: [] } },
+      ];
+      const result = invertFacilityUseItems(items);
+      expect(result).to.equal(items);
+    });
+
+    it('should invert FacilityUse items and concatenate them with other items', () => {
+      // Test Objects
+      const facilityUseItems = [
+        { 
+          state: 'updated',
+          data: { 
+            '@context': 'https://openactive.io/',
+            name: 'Facility Use 1',
+            '@type': 'FacilityUse', 
+            individualFacilityUse: [{ '@type': 'IndividualFacilityUse', '@id': '1' }] 
+          } 
+        },
+        { 
+          state: 'updated',
+          data: { 
+            name: 'Facility Use 2',
+            '@context': 'https://openactive.io/',
+            '@type': 'FacilityUse', 
+            individualFacilityUse: [{ '@type': 'IndividualFacilityUse', '@id': '2' }] 
+          } 
+        },
+      ];
+      const otherItems = [
+        { data: { '@type': 'FacilityUse', someData: 'value' } },
+      ];
+      const items = facilityUseItems.concat(otherItems);
+
+      // Test
+      const result = invertFacilityUseItems(items);
+
+      // Assertions
+      expect(result).to.have.lengthOf(3);
+
+      expect(result[0]).to.have.property('id', '1');
```

Comment:
Sorry I know this is outside of the scope of your changes but do you happen to know why this function changes the RPDE ID to match the IFU's DATA ID? That seems like it could cause issues

Reply:
To simulate the subEvents coming from a separate feed, each item will need a unique RPDE ID right? So I think it just uses the data ID so that an RPDE ID doesn't have to made up maybe?